src/base/emu-i386/simx86/codebuf.c

```diff
@@ -0,0 +1,100 @@
+/*
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+
+/*
+ * Purpose: dlmalloc's memory spaces support
+ * and circumvent W^X restrictions.
+ *
+ * Author: @stsp
+ *
+ */
+#include <stddef.h>
+#include <assert.h>
+#include <sys/mman.h>
+#include "emu86.h"
+#include "misc/dlmalloc.h"
+
+#ifndef MAP_JIT
+#define MAP_JIT 0
+#endif
+#define CODEBUF_SZ (128 * 1024 * 1024)
+static mspace cspace;
+static unsigned char *wbase;
+static unsigned char *xbase;
+
+void InitGenCodeBuf(void)
+{
+    void *addr;
+
+#if defined(HAVE_MREMAP)
```

Comment:
```suggestion
#if HAVE_DECL_MREMAP_MAYMOVE
```
(not `#ifdef` as it's set to 0 on Mac) There's a weird `-Dlinux` in Makefile that causes dlmalloc.h to define HAVE_MREMAP. In any case you also need MREMAP_MAYMOVE, so it seems to me that HAVE_DECL_MREMAP_MAYMOVE is more appropriate.

With that it compiles and runs on Mac (though I'd have to add the mach call to get w^x there)

Reply:
Done.